Modules/_datetimemodule.c

```diff
@@ -5514,19 +5517,27 @@ datetime_utcfromtimestamp(PyObject *cls, PyObject *args)
 static PyObject *
 datetime_strptime(PyObject *cls, PyObject *args)
 {
-    static PyObject *module = NULL;
     PyObject *string, *format;
 
     if (!PyArg_ParseTuple(args, "UU:strptime", &string, &format))
         return NULL;
 
+    PyObject *result = NULL;
+    PyObject *current_mod = NULL;
+    datetime_state *st = GET_CURRENT_STATE(current_mod);
+    PyObject *module = st->strptime;
     if (module == NULL) {
         module = PyImport_ImportModule("_strptime");
```

Comment:
Would it be much slower if we were to always import it, rather than caching it in module state?  That would certainly be simpler.

Reply:
```py
from timeit import timeit
cmd = 'import _datetime; f = _datetime.datetime.strptime'
print(timeit('f("200001", "%Y%m")', cmd))

(a) 8.646  # main (local static)
(b) 8.873  # module state
(c) 9.070  # PyImport_Import(&_Py_ID(_strptime))
(d) 9.544  # PyImport_ImportModule("_strptime")
```

Measured.  (c) has been applied from 12fad4c4d09908f52f1c1a98344bcd9d55580f9f.

Reply:
Tried backporting (Win non-debug):

```py
3.12
(a) 7.820 (base)  3.12.4+ (local static)
(c) 8.568 (1.09x) PyImport_Import(&_Py_ID(_strptime))
(d) 9.073 (1.16x) PyImport_ImportModule("_strptime")

3.11 
(a) 7.901 (base)  3.11.9+ (local static)
(c) 8.524 (1.08x) PyImport_Import(&_Py_ID(_strptime))
(d) 8.774 (1.11x) PyImport_ImportModule("_strptime")
```

Reply:
Yeah (c) looks good.